cpp/test/sg/fil_test.cu

```diff
@@ -748,6 +756,19 @@ std::vector<FilTestParams> predict_dense_inputs = {
   FIL_TEST_PARAMS(num_rows = 103, num_cols = 100'000, depth = 5, num_trees = 1,
                   algo = BATCH_TREE_REORG, leaf_algo = CATEGORICAL_LEAF,
                   num_classes = 3),
+  FIL_TEST_PARAMS(algo = BATCH_TREE_REORG, threads_per_tree = 2),
+  FIL_TEST_PARAMS(algo = BATCH_TREE_REORG, threads_per_tree = 4),
+  FIL_TEST_PARAMS(algo = BATCH_TREE_REORG, threads_per_tree = 8),
+  FIL_TEST_PARAMS(algo = BATCH_TREE_REORG, threads_per_tree = 16),
+  FIL_TEST_PARAMS(algo = BATCH_TREE_REORG, threads_per_tree = 32),
```

Comment:
Consider reducing the number of tests for `BATCH_TREE_REORG` and adding more tests for other values of `algo`.